### DISCUSSION TITLE
(next-urql): undefined context in some requests

### DISCUSSION THREAD

**Discussion by lightify97** [urql-graphql/urql#2550] [Q&A] [answered]
I'm using next-urql with SSR enabled. The isssue is that the context passed at each request comes out to be undefined for some requests.

Client:
```ts
export const createUrqlClient = (ssrExchange: any, ctx: any) => {
  let cookie = '';
  if (isServer()) {
    console.dir(typeof ctx, { depth: 1 });
  }
  return {
    url: 'http://localhost:4000/graphql',
    fetchOptions: {
      credentials: 'include' as const,
      headers: cookie ? { cookie } : undefined,
    },
    ...
    ...
```

Console Logs:
```
'object'
'undefined'
'undefined'
'object'
'undefined'
'undefined'
wait  - compiling...
event - compiled client and server successfully in 1064 ms (2906 modules)
wait  - compiling / (client and server)...
wait  - compiling...
event - compiled client and server successfully in 462 ms (2954 modules)
'object'
'undefined'
'undefined'
wait  - compiling...
event - compiled client and server successfully in 1165 ms (2906 modules)
wait  - compiling / (client and server)...
wait  - compiling...
event - compiled client and server successfully in 345 ms (2954 modules)
'object'
'undefined'
'undefined'
wait  - compiling...
event - compiled client and server successfully in 681 ms (2906 modules)
wait  - compiling / (client and server)...
wait  - compiling...
event - compiled client and server successfully in 289 ms (2954 modules)
'object'
'undefined'
'undefined'
wait  - compiling...
event - compiled client and server successfully in 955 ms (2906 modules)
wait  - compiling / (client and server)...
wait  - compiling...
event - compiled client and server successfully in 314 ms (2954 modules)
'undefined'
'undefined'
wait  - compiling...
event - compiled client and server successfully in 882 ms (2906 modules)
wait  - compiling / (client and server)...
wait  - compiling...
event - compiled client and server successfully in 280 ms (2954 modules)
'object'
'undefined'

```

Basically I need to pass browser cookie to my graphql server while server rendering the page. Hope I put my point across correctly.

**Comment by lightify97** [comment]
@kitten Any help please..

**Comment by JoviDeCroock** [comment, marked as answer]
It's Sunday and this isn't our job so is an issue that is less than 24 hours old really worth tagging people in? 😅 

That being said, this makes sense, we only have access to `context` during the prepass phase https://github.com/FormidableLabs/urql/blob/main/packages/next-urql/src/with-urql-client.ts#L129 and not during the render-to-string phase https://github.com/FormidableLabs/urql/blob/main/packages/next-urql/src/with-urql-client.ts#L77 this however does mean that as long as you use it in that first phase it will work because the second phase does not fetch data and synchronously gets the stringified html content. Not entirely sure whether you tried to implement it yet but it should just work and on the client-side it will work through doing that `include` there

**Comment by kitten** [comment]
Just here to say, yea, this is really disrespectful of our time, and don't get this the wrong way; we're taking this opportunity to educate you about this, so you won't do this on other GitHub repositories either. We're doing this work voluntarily and are basically donating free time. In this case, I'm not even the right person to answer this question as I haven't worked on `next-urql`, so please wait for us to get to issues as we see fit. Usually this means as soon as possible anyway

**Comment by lightify97** [comment]
@kitten Sorry for any inconvenience . I just found your answers really helpful on a lot of other issues that's why I mentioned you. I'll be careful next time.

**Comment by JoviDeCroock** [comment]
@lightify97 was your issue resolved after https://github.com/FormidableLabs/urql/issues/2466#issuecomment-1140490166?

**Comment by lightify97** [comment]
No, it's not.
